core/libs/editor/src/lib/editor-toolbar/components/generate-open-api/generate-open-api.component.ts

```diff
@@ -31,88 +32,216 @@ export interface OpenApi {
 
-  generateOpenApiSpec(): void {
-    const openApi: OpenApi = {
-      output: this.getControlValue('output') as string,
-      baseUrl: this.getControlValue('baseUrl') as string,
-      includeQueryApi: this.getControlValue('includeQueryApi') as boolean,
-      useSemanticVersion: this.getControlValue('useSemanticVersion') as boolean,
-      paging: this.getControlValue('paging') as string,
-      language: this.getControlValue('language') as string,
-    };
+  private setupFormListeners(): void {
+    this.subscriptions.add(this.output?.valueChanges.subscribe(() => this.removeUploadedFile()));
+
+    this.subscriptions.add(
+      this.activateResourcePath?.valueChanges.subscribe(activateResourcePath => {
+        const resourcePathControl = this.form.get('resourcePath');
+        const fileControl = this.form.get('file');
+
+        if (activateResourcePath) {
+          resourcePathControl?.setValue('/resource/{resourceId}');
+          resourcePathControl?.setValidators(this.resourcePathValidators);
+          fileControl?.setValidators(Validators.required);
+        } else {
+          resourcePathControl?.setValue(null);
+          resourcePathControl?.setValidators(null);
+          fileControl?.setValidators(null);
+        }
+
+        resourcePathControl?.updateValueAndValidity();
+        fileControl?.updateValueAndValidity();
+      }),
+    );
+  }
+
+  @HostListener('dragover', ['$event'])
+  preventDragDefaults(event: Event): void {
+    event.preventDefault();
+  }
+
+  @HostListener('drop', ['$event'])
+  handleFileDrop(event: DragEvent): void {
+    event.preventDefault();
+    if (!this.dropArea.nativeElement.contains(event.target)) return;
+
+    const files = event.dataTransfer?.files;
+    if (files && files.length) {
+      const file = files[0];
+      if (this.validateFile(file)) {
+        this.processFile(file);
+      } else {
+        this.showError();
+      }
+    }
+  }
 
+  private validateFile(file: File): boolean {
+    const fileType = this.form.value.output;
+    return fileType === 'json' ? file.name.endsWith('.json') : file.name.endsWith('.yaml') || file.name.endsWith('.yml');
+  }
+
+  private processFile(file: File): void {
+    this.uploadedFile = file;
+    this.form.patchValue({file: file});
+    this.readFileContent(file);
+  }
+
+  private readFileContent(file: File): void {
+    const reader = new FileReader();
+    reader.onload = () => this.handleFileContent(file, reader.result as string);
+    reader.readAsText(file);
+  }
+
+  private handleFileContent(file: File, content: string): void {
+    const fileType = this.getFileType(file);
+    const propertyName = fileType === 'json' ? 'jsonProperties' : 'ymlProperties';
+    this.form.patchValue({[propertyName]: content});
+  }
+
+  private getFileType(file: File): 'json' | 'yml' {
+    if (file.name.endsWith('.json')) {
+      return 'json';
+    } else if (file.name.endsWith('.yaml') || file.name.endsWith('.yml')) {
+      return 'yml';
+    }
```

Comment:
maybe we can have an enum with 'yml', 'yaml' and json

Reply:
I have considered the abstraction, but given that its current application is limited only to our immediate needs, I recommend not implementing it at this stage.